### DISCUSSION TITLE
Список всех файлов в системе

### DISCUSSION THREAD

**Discussion by mihail-cornev** [totumonline/totum-ru-issues-and-discussions#289] [Как это сделать?] [answered]
Подскажите пожалуйста решение. Есть идея сделать таблицу Файлы в Тотум, куда изначально будут загружаться файлы по типу - 1 строка - 1 файл. Нужно это для того, чтобы для каждого файла можно было добавить разные характеристики и потом проще искать, и чтобы добавлять в разные части системы один файл, без дублирования, а так же чтобы удалять файлы (чистить систему), или например найти все файлы, которые прислал определённый клиент и т.п.
Как выводить изображения по селекту в другие таблицы? Было бы удобно "показывать файл как": картинку, значок, текст и т.п.
Стоит ли этим заморачиваться? Пробовал ли кто-то такое реализовывать?

**Comment by ttm-alexeypolunin** [comment]
Я позже на этой неделе напишу свои соображения. Можно хранить в отдельной таблице, мы так например делаем в документации.

**Reply by mihail-cornev** [reply]
Хочу вернуться к обсуждению данной темы, может получится найти какое-то решение. задача сделать единую таблицу со всеми файлами, чтобы можно было видеть кто когда и что загружал, чтобы не дублировались файлы в системе, чтобы можно было использовать один и тот же файл в разных частях системы.

Видео примерно как хочется чтобы это было. [Файлы — .webm](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/113771444/73b23d79-2db3-4a69-9887-016f55cf0343)

Пример:
Имеем таблицу ФАЙЛЫ и таблицу циклов ЗАКАЗЫ

загружаем из т. ЗАКАЗЫ через временную таблицу файл.
этот файл загружается в т.ФАЙЛЫ.
но при этом в т.ЗАКАЗЫ мы можем открыть файл или скачать.

UPD: возникают ещё вопросы:
- как отправлять эти файлы через email, или как добавлять их в уведомления через telegram

**Reply by ttm-alexeypolunin** [reply]
1. Снимаете галочку Копировать файл при дублировании
2. В заказе кодом берете содержимое поля файл
3. Для редактирования удаленно используете linkToEdit
4. Временная -> Простая здесь: https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/276
5. Отправка в email — смотрите документацию по emailSend
6. Как отправить файл в telegram —> смотреть в документации telegram

**Reply by mihail-cornev** [reply]
> 2. В заказе кодом берете содержимое поля файл

непонятно как это работает.

в таблице ЗАКАЗЫ я делаю расчётную таблицу Файлы, в ней я делаю после селект по таблице Файлы.
и рядом поле Файл, которым пытаюсь кодом взять файл, но ничего не происходит.

![Screenshot_4](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/113771444/0bdc161c-7583-4a1e-9f73-a971d2e9103b)

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Да, можно держать файлы в центральной таблице. Внутрь циклов показывать их как `tab_`. Но это не точно, тк у нас там возможно есть ошибки связанные с Защищенными файлами — надо пробовать.

На сервере все файлы хранятся в папке `fls/HOST`. Защищенные файлы отдельно. Имена у них на диске уникальные от `name_таблицы_name_поля_id_строки`.

Поэтому можно сделать несколько полей Файл ссылающихся на один и тот-же файл на диске.

**Reply by mihail-cornev** [reply]
Подскажите пожалуйста если можно такое сделать:

Есть таблица ФАЙЛЫ в которую мы загружем все файлы, один файл - одна строка.
Есть другая таблица: МОИ ФАЙЛЫ. В ней есть два поля ФАЙЛ и ИЗОБР. Поле файл с типом селект выбираем из таблицы ФАЙЛЫ нужный файл - тут всё просто. А как вывести изображение или иконку выбранного значения в поле ИЗОБР?

И ещё вопрос, как сделать так, чтобы окно загрузки файлов сразу открывалось и не приходилось дополнительно нажимать на кнопку "добавить файл" внутри открывшегося окна?

![Screenshot_9](https://user-images.githubusercontent.com/113771444/211591636-016a3fcc-17ad-467f-a03a-9bf1915f6a7e.png)

**Reply by ttm-alexeypolunin** [reply]
Опишите подробнее про поля Файл и Изображение? Это в Хедере внутри цикла?
